pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1658,10 +1665,12 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(ctx context.Context, clusterName string
 					listenerCreateOpt.InsertHeaders = map[string]string{"X-Forwarded-For": "true"}
 				}
 
-				listenerCreateOpt.TimeoutClientData = &timeoutClientData
-				listenerCreateOpt.TimeoutMemberData = &timeoutMemberData
-				listenerCreateOpt.TimeoutMemberConnect = &timeoutMemberConnect
-				listenerCreateOpt.TimeoutTCPInspect = &timeoutTCPInspect
+				if openstackutil.IsOctaviaFeatureSupported(lbaas.lb, openstackutil.OctaviaFeatureTimeout) {
+					listenerCreateOpt.TimeoutClientData = &timeoutClientData
+					listenerCreateOpt.TimeoutMemberData = &timeoutMemberData
+					listenerCreateOpt.TimeoutMemberConnect = &timeoutMemberConnect
+					listenerCreateOpt.TimeoutTCPInspect = &timeoutTCPInspect
+				}
 
```

Comment:
Can we remove the check here? `EnsureLoadBalancer` is now only for Neutron-LBaaS which doesn't support version query, you have already added the check for Octavia in `ensureOctaviaListener`.

Reply:
oh okay.. I will change that

Reply:
in that case these functions could be cleaned little bit, but I will leave that to next pr